daemon/container_linux.go

```diff
@@ -0,0 +1,20 @@
+//+build !windows
+
+package daemon
+
+import (
+	containertypes "github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/container"
+)
+
+func (daemon *Daemon) setSecurityOptions(container *container.Container, hostConfig *containertypes.HostConfig) error {
+	container.Lock()
+	defer container.Unlock()
+	if err := parseSecurityOpt(container, hostConfig); err != nil {
+		return err
+	}
+	if daemon.apparmorEnabled && container.AppArmorProfile == "" {
+		container.AppArmorProfile = "docker-default"
```

Comment:
Also need to handle `--privileged`. See https://github.com/docker/docker/blob/master/daemon/oci_linux.go#L709-L714

Even with `privileged` fixed this is a slight behavioral change. If apparmor is enabled after container was created this var will not change. I think if this is only called on inspecting single container we can lazy load the value like we do in container startup.